### PR TITLE
Remove unimplemented virus scanning hooks

### PR DESCRIPTION
**Pull request by freqlabs**
### Motivation and Context

Code cleanup

### Description

Remove code that doesn't do anything.

### How Has This Been Tested?

ZTS ran with PR

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Review by adamdmoss — APPROVED** [review]

File: module/os/freebsd/zfs/zfs_vnops_os.c [hunk 1]
```diff
@@ -231,15 +231,6 @@ zfs_open(vnode_t **vpp, int flag, cred_t *cr)
 		return (SET_ERROR(EPERM));
 	}
 
-	if (!zfs_has_ctldir(zp) && zp->z_zfsvfs->z_vscan &&
```
Comment (behlendorf) [inline module/os/freebsd/zfs/zfs_vnops_os.c:234]: We should be able to remove the `z_vscan` member form the `zfsvfs_t`.

**Review by behlendorf — COMMENTED** [review]
Glad to see this dead code finally removed.  I know it's never been used on Linux, I'm not sure it was even ever used on Solaris.

**Comment by freqlabs** [comment]
* Removed more unused code for vscan
* Updated zfsprops(8) to say this property is unused by OpenZFS on any platform, not just Linux

**Comment by behlendorf** [comment]
@ikozhukhov @jasonbking @tsoome I was hoping one of you might be able to shed some light on if we really need to keep this for illumos?

**Comment by freqlabs** [comment]
> i have no ideas about illumos, but we have to keep it for DilOS

DilOS doesn't need the FreeBSD and Linux platform code to be there. With this platform code removed, nothing in the tree at this time uses the two defines I'm removing from common code. You should be able to keep those in DilOS and merge them back in with the code that uses it when the time comes without any difficulty.

**Comment by ikozhukhov** [comment]
> > i have no ideas about illumos, but we have to keep it for DilOS
> 
> DilOS doesn't need the FreeBSD and Linux platform code to be there. With this platform code removed, nothing in the tree at this time uses the two defines I'm removing from common code. You should be able to keep those in DilOS and merge them back in with the code that uses it when the time comes without any difficulty.

we are talking about shared codebase between platforms and will be much more better isolate some another platform code to `#ifdef <platform>` instead of remove it at all. in this case you try to talk about - use your fork, instead of how to do it better. it's a pity yo hear it again and again.

**Comment by freqlabs** [comment]
I don't understand what you want to `#ifdef` here. I'm only removing two `#define` lines from the shared code, and a comment.

File: include/sys/dsl_deleg.h [hunk 2]
```diff
@@ -48,7 +48,6 @@ extern "C" {
 #define	ZFS_DELEG_PERM_RECEIVE		"receive"
 #define	ZFS_DELEG_PERM_ALLOW		"allow"
 #define	ZFS_DELEG_PERM_USERPROP		"userprop"
-#define	ZFS_DELEG_PERM_VSCAN		"vscan"
```
Comment (ikozhukhov) [inline include/sys/dsl_deleg.h:51]: it is example for future use on DilOS and can be done on others places in shared code
```
#ifdef __dilos__
#define	ZFS_DELEG_PERM_VSCAN		"vscan"
#endif
```

**Review by behlendorf — APPROVED** [review]
Dropping just the unused platform specific code makes sense to me.  New platform specific code can be added when a platform which does support this is added to OpenZFS.

File: cmd/zfs/zfs_main.c [hunk 3]
```diff
@@ -5518,11 +5518,6 @@ deleg_perm_comment(zfs_deleg_note_t note)
 	case ZFS_DELEG_NOTE_CHANGE_KEY:
 		str = gettext("Allows changing or adding an encryption key");
 		break;
-/*
- *	case ZFS_DELEG_NOTE_VSCAN:
- *		str = gettext("");
- *		break;
- */
```
[the remark this answers is not in the document]
Comment (freqlabs) [reply cmd/zfs/zfs_main.c:5525]: It's a comment, I figured any platform that implements vscan would have to remove these lines anyway.

[the remark this answers is not in the document]
Comment (ikozhukhov) [reply cmd/zfs/zfs_main.c:5525]: i'd like leave it as is for now and update later is needed.

[the remark this answers is not in the document]
Comment (behlendorf) [reply cmd/zfs/zfs_main.c]: Oh, my eyes completely missed that it was commented out!  I'm fine with it either way then.  LGTM.

**Pull request merged**